atom/src/catom.cpp

```diff
@@ -448,11 +445,8 @@ CAtom_getstate( CAtom* self )
 }
 
 PyObject*
-CAtom_setstate( CAtom* self, PyObject* args )
+CAtom_setstate( CAtom* self, PyObject* state )
 {
-    if( PyTuple_GET_SIZE( args ) != 1 )
-        return cppy::type_error( "__setstate__() takes exactly one argument" );
-    PyObject* state = PyTuple_GET_ITEM( args, 0 );
     cppy::ptr itemsptr = PyMapping_Items(state);
```

Comment:
Do we need a `PyMapping_Check` before this?

That said, `PyMapping_Check` doesn't even check for `items()`.

Should we force it to be a `dict`?

Some of the below logic is questionable and may need some attention. I don't think that `__setstate__` should trigger observers, which means it shouldn't go thru the `__setattr__` protocol.